app/src/main/java/org/example/viotester/modules/MapOverlayActivity.java

```diff
@@ -0,0 +1,442 @@
+//        try {
+//            String readLine = null;
+//            while ((readLine = br.readLine()) != null) {
+//                String[] row = readLine.split(",");
+//                gpsRoute.addPoint(
+//                        new LatLng(Float.parseFloat(row[1]), Float.parseFloat(row[2])),
+//                        Double.parseDouble(row[0]) - 200.);
+//            }
+//            is.close();
+//            br.close();
+//        } catch (IOException e) {
+//        }
+//        gpsRoute.update();
+//
+//        is = this.getResources().openRawResource(R.raw.track);
+//        br = new BufferedReader(new InputStreamReader(is));
+//        try {
+//            String readLine = null;
+//            while ((readLine = br.readLine()) != null) {
+//                String[] row = readLine.split(",");
+//                double time = Double.parseDouble(row[0]) - 200.;
+//                double x = Double.parseDouble(row[1]);
+//                double y = Double.parseDouble(row[2]);
+//                LatLng newPosition = enuToWgs(gpsRoute.first(), new Point(x, y));
+//                trackingRoute.addPoint(newPosition, new Point(-x, y, time));
+//            }
+//            is.close();
+//            br.close();
+//        } catch (IOException e) {
+//        }
+//        trackingRoute.updateAligned(gpsRoute);
+//
+//        updateCameraPosition();
+//    }
+
+    private void updateCameraPosition() {
+        final int paddingPixels = 50;
+        LatLngBounds.Builder b = new LatLngBounds.Builder();
+        if (gpsRoute.coords.size() > 0) {
+            b.include(gpsRoute.coords.get(0));
+            b.include(gpsRoute.coords.get(gpsRoute.coords.size() - 1));
+        }
+        if (trackingRoute.coords.size() > 0) {
+            b.include(trackingRoute.coords.get(0));
+            b.include(trackingRoute.coords.get(trackingRoute.coords.size() - 1));
+        }
+        googleMap.moveCamera(CameraUpdateFactory.newLatLngBounds(
+                b.build(),
+                paddingPixels));
+//        final float ZOOM_LEVEL = 20.f;
+//        googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(
+//                gpsRoute.coords.get(gpsRoute.coords.size() - 1), ZOOM_LEVEL));
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        handler.postDelayed(runnable = new Runnable() {
+            public void run() {
+                handler.postDelayed(runnable, TRACKING_POLL_INTERVAL_MS);
+                double[] pose = mAlgorithmWorker.getPose();
+                if (pose != null && pose[0] > 0.0) {
+                    onPose(pose);
+                }
+            }
+        }, TRACKING_POLL_INTERVAL_MS);
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        handler.removeCallbacks(runnable);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        handler.removeCallbacks(runnable);
+    }
```

Comment:
Note that if you push from the AlgorithmWorker instead of polling here (see other comment for details), this stuff is unnecessary